lib/pos_blockchain.go

```diff
@@ -407,6 +382,29 @@ func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock) error {
 	return nil
 }
 
+// tryApplyNewTip attempts to apply the new tip to the best chain. It will do the following:
+// 1. Check if we should perform a reorg. If so, it will handle the reorg. If reorging causes an error, return false and error.
+// 2. Check if the incoming block extends the chain tip after reorg. If not, return false and nil
+// 3. If the incoming block extends the chain tip, we can apply it by calling addBlockToBestChain. Return true and nil.
+func (bc *Blockchain) tryApplyNewTip(desoBlock *MsgDeSoBlock, currentView uint64, lineageFromCommittedTip []*BlockNode) (_appliedNewTip bool, _err error) {
+	if _, err := bc.tryReorgToNewTip(desoBlock, currentView, lineageFromCommittedTip); err != nil {
```

Comment:
In this context, the `tryReorgToNewTip` name gives the impression that the chain is being reorged and the new tip is being applied internally within `tryReorgToNewTip`. When really, it's just the ancestors that are being reorged.

I suggest just merging the implementations of the `tryApplyNewTip` and `tryReorgToNewTip` functions into a single `tryApplyNewTip` function given how related the entire code is. I admittedly found it a bit difficult to follow what the state of the `bestChain` was and why after this operation.